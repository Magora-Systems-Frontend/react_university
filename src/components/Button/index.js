import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import './style.scss';

export default class Button extends Component {
  static propTypes = {
    children: PropTypes.any,
    onClick: PropTypes.func,
    type: PropTypes.string,
    modificationClass: PropTypes.string,
  };

  render() {
    const { children, onClick, colorStyle, cssModify, type, modificationClass } = this.props;
    const customClass = {
      bordered: 'btn_bordered',
      colored: 'btn_colored',
      transparent: 'btn_transparent',
      search: 'btn_search',
      mobileLogin: 'btn_mobile-login',
      category: 'btn_category-toggle',
    };
    return (
      <button
        className={cn('btn-action', customClass[colorStyle], modificationClass)}
        onClick={onClick ? onClick : () => {}}
        style={cssModify}
        type={type}>
        {children}
      </button>
    );
  }
}
