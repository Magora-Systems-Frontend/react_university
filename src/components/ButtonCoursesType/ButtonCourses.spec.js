import React from 'react';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Button from '../Button';
import { ButtonCoursesType as Component } from './index.js';

configure({ adapter: new Adapter() });

describe('component snapshot', () => {
  it('capturing snapshot of buttons', () => {
    const renderedValue = renderer.create(<Component clickButton={() => {}} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('Shallow render comonent', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Component />);
  });

  it('check count Buttons', () => {
    expect(wrapper.find(Button)).toHaveLength(4);
  });
});
