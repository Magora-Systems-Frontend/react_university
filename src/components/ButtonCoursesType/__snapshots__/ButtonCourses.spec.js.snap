// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component snapshot capturing snapshot of buttons 1`] = `
<div
  className="button-container"
>
  <button
    className="btn-action btn_category-toggle btn_category-toggle_active"
    onClick={[Function]}
  >
    Development
  </button>
  <button
    className="btn-action btn_category-toggle"
    onClick={[Function]}
  >
    Design
  </button>
  <button
    className="btn-action btn_category-toggle"
    onClick={[Function]}
  >
    Business
  </button>
  <button
    className="btn-action btn_category-toggle"
    onClick={[Function]}
  >
    Photography
  </button>
</div>
`;
